demo/hello-world/run.py

```diff
@@ -1,42 +1,248 @@
+                    "name": "file",
+                    "config": {
+                        "path": filepath,
+                    }
+                },
+                "format": {
+                    "name": "csv",
+                    "config": {}
+                }
+            }
+        }
+        if name not in existing_connectors:
+            response = requests.post(f"{api_url}/v0/connectors", json=connector_json)
+        else:
+            response = requests.patch(f"{api_url}/v0/connectors/{existing_connectors[name]}", json=connector_json)
+        response.raise_for_status()
+
+        # Add to list of connectors
+        connectors.append({
+            "connector_name": name,
+            "is_input": is_input,
+            "name": name,
+            "relation_name": stream
+        })
+
+    # Pipeline
+    pipeline_name = "demo-hello-world-pipeline"
+    pipeline_id = create_start_pipeline(
+        api_url,
+        {},
+        program_name,
+        pipeline_name,
+        num_workers=6,
+        connectors=connectors
+    )
+
+    # Wait till the pipeline is completed
+    while not requests.get(f"{api_url}/v0/pipelines/{pipeline_id}/stats").json()["global_metrics"]["pipeline_complete"]:
+        time.sleep(1)
+    print("Pipeline finished")
+
+    # Verify result
+    with open(MATCHES_CSV) as f:
+        content = f.read().strip()
+        assert content == "Hello world!,1"
+    print("Demo success: hello-world")
+
+
+###############################################################################
+# HELPER FUNCTIONS BELOW
+
+
+def create_compile_program(
+        api_url: str,
+        headers: dict,
+        program_sql: str,
+        program_name: str,
+        program_description: str = ""
+) -> str:
+    """
+    Create and compile a SQL program for a Feldera pipeline.
+    Function version: 0.0.1
+
+    :param api_url:               URL of the Feldera REST API (e.g., "http://localhost:8080")
+    :param headers:               Headers to add to each HTTP request
+    :param program_sql:           Program SQL
+    :param program_name:          Program name
+    :param program_description:   (Optional) Program description
+
+    :return: Program identifier
+    """
+    program = {
+        "name": program_name,
+        "description": program_description,
+        "code": program_sql,
+    }
+
+    # Patch existing program if exists, else post
+    print(f"Creating program named {program['name']}...")
+    existing_programs = {
+        entry["name"]: entry["program_id"]
+        for entry in requests.get(api_url + "/v0/programs", headers=headers).json()
+    }
+    if program["name"] in existing_programs:
+        print(f"Program {program['name']} already exists, instead patching")
+        program_id = existing_programs[program["name"]]
+        patch_response = requests.patch(f"{api_url}/v0/programs/{program_id}",
+                                        json=program, headers=headers)
```

Comment:
Could you rebase this over the names-in-urls branch and send the PR to that branch?

Reply:
Done :+1: